### PR TITLE
Add tx based transfer support to archiving

### PR DESCRIPTION
**Pull request by nekia**
Signed-off-by: Atsushi Neki <atsushin@fast.au.fujitsu.com>

#### Type of change

- New feature

#### Description

* This change adds the support for transaction based data transferring from block respository to client. When incur the access to transaction data which has already been archived, client peer can dispatch the access request toward archiver peer (repository peer) to retrieve the requested transaction data.

#### Additional details

* Currently this feature has been implemented with using existing logic for state request. Ultimately we'll separate our implementation from the existing core logic.
* We also need to update [fabric-proto repository](https://github.com/hyperledger/fabric-protos) for adding new message to gossip
* Added new subcommands to `verify` command of peer. We can use them for testing purpose.

### CODE REVIEW

File: core/ledger/ledgerstorage/store.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ import (
 	"github.com/hyperledger/fabric/core/ledger/pvtdatastorage"
 )
 
-const maxBlockFileSize = 64 * 1 * 1024
+const maxBlockFileSize = 16 * 1 * 1024
```
Comment (anand-jpa) [inline core/ledger/ledgerstorage/store.go:23]: Why are we reducing the max block file size from 64kb to 16kb?

Comment (nekia) [reply core/ledger/ledgerstorage/store.go:23]: It's just for testing purpose. We'll revert it later on.

Comment (anand-jpa) [reply core/ledger/ledgerstorage/store.go:23]: ok

**Review by anand-jpa — APPROVED** [review]
Good

**Pull request merged**